lib/elixir/lib/protocol.ex

```diff
@@ -720,7 +720,9 @@ defmodule Protocol do
               unquote(__MODULE__.Any).__impl__(:target)
             rescue
               UndefinedFunctionError ->
-                nil
+                reraise Protocol.UndefinedError,
+                        [protocol: unquote(__MODULE__), value: :how_do_i_pass_this_value?],
```

Comment:
I tried for quiet some time to pass value to here to no avail. What' s the way to solve this?

Reply:
Right, because here is not really the correct place to raise. This could rather return `nil` so the parent raises it but as mentined, it would be a semantic change on behaviour.

Reply:
I converted any_impl_for to an anonymous function of arity-1. The trouble that i have when it is currently called `unquote(any_impl_for)`  I tried to need to unquote `data` inside another `unquote` call, ending up with `unquote(any_impl_for.(unquote(data))`).

The other solution is modify `Protocol.UndefinedError` to accept `:message` and pass only this attribute.

Reply:
Right. The other thing to keep in mind is that this code is very performance sensitive. Wrapping the call in a try/catch can negatively affect performance. I think it is best to drop this. :/

Reply:
How about something like this?
```elixir
      any_impl_for =
        if @fallback_to_any do
          quote do
            if function_exported?(unquote(__MODULE__.Any), :__impl__, 1) do
              unquote(__MODULE__.Any).__impl__(:target)
            else
              raise Protocol.UndefinedError,
                message: "protocol #{inspect(__MODULE__)} has no implementation for Any"
            end
          end
        else
          nil
        end
```

Reply:
It is not enough because the module may not have been loaded yet. You would need to add a Code.ensure_loaded? check which, once again, can be expensive. Given the performance concern and change of behaviour, we should drop this. Thank you!

Reply:
Fair enough.